packages/smooth_app/lib/pages/preferences/user_preferences_settings.dart

```diff
@@ -79,29 +82,70 @@ class UserPreferencesSettings extends AbstractUserPreferences {
             ],
           ),
         ),
-        Padding(
-          padding: const EdgeInsets.symmetric(horizontal: LARGE_SPACE),
-          child: Text(
-            appLocalizations.main_app_color,
-            style: themeData.textTheme.headline4,
-          ),
-        ),
-        Padding(
-          padding: const EdgeInsets.symmetric(
-            horizontal: LARGE_SPACE,
-            vertical: VERY_SMALL_SPACE,
-          ),
-          child: Wrap(
-            spacing: 8.0,
-            children: List<Widget>.generate(
-              _ORDERED_COLOR_TAGS.length,
-              (final int index) => _getColorButton(
-                themeData.colorScheme,
-                _ORDERED_COLOR_TAGS[index],
-                themeProvider,
-              ),
-            ),
-          ),
+        StatefulBuilder(
+          builder:
+              (BuildContext context, void Function(VoidCallback fn) setState) {
+            bool shouldColorThemeShown = true;
+
+            final ThemeMode themeMode =
+                context.select<ThemeProvider, ThemeMode>(
+                    (ThemeProvider value) => value.currentThemeMode);
+
+            switch (themeMode) {
+              case ThemeMode.light:
+                shouldColorThemeShown = true;
+                break;
+              case ThemeMode.dark:
+                shouldColorThemeShown = false;
+                break;
+              case ThemeMode.system:
+                final SingletonFlutterWindow window =
+                    WidgetsBinding.instance.window;
+
+                shouldColorThemeShown =
+                    window.platformBrightness == Brightness.light;
+
+                // This callback is called every time the brightness changes.
+                window.onPlatformBrightnessChanged = () {
+                  setState(() {});
+                };
+                break;
+            }
+
+            if (!shouldColorThemeShown) {
+              return const SizedBox.shrink();
```

Comment:
I'm not a big fan of `SizedBox.shrink()`. The clean solution would obviously be not to add an item to the list, but that would change the rest of the method too.
Let's say that's good enough.